src/AllArticles.jsx

```diff
@@ -1,24 +1,31 @@
 import ArticleThumbnail from "./Article-thumbnail";
 import { getAllArticles } from "../apis";
-
 import { useEffect, useState } from "react";
+import Loading from "./Loading"
 
 function AllArticles() {
-    const [allArticles, setAllArticles] = useState([]);
+  const [allArticles, setAllArticles] = useState([]);
+  const [loading, setLoading]=useState(true)
     
     useEffect(() => {
       getAllArticles().then(({articles}) => {
         setAllArticles(articles)
-      });
+        setLoading(false)
+      }).catch((err) => {
+        console.log(err);
```

Comment:
I guess here is an opportunity to tell your user that theres been an issue fetching the articles, it's unlikely the average user knows how to check the console, let alone when using a mobile app :)